dvc/api.py

```diff
@@ -0,0 +1,24 @@
+from dvc.utils.compat import open as _open
+from dvc.repo import Repo
+
+
+def get_url(path, repo_dir=None, remote=None):
+    """Returns an url of `path` in default or specified remote"""
+    repo = Repo(repo_dir)
+    out, = repo.find_outs_by_path(path)
+    return str(repo.get_remote(remote).checksum_to_path_info(out.checksum))
+
+
+def open(path, repo_dir=None, remote=None):
+    """Opens a specified resource as a file descriptor"""
+    repo = Repo(repo_dir)
+    out, = repo.find_outs_by_path(path)
+    if out.isdir():
+        raise ValueError("Can't open a dir")
+
+    with repo.state:
```

Comment:
Here API is really using Repo internals(e.g. you've renamed collect_used_cache to make it public, but it shouldn't be exposed actually), I would rather move this into `dvc/repo/open.py` and have
```
def open(path, repo_dir=None, remote=None):
    return Repo(repo_dir).open(path, remote=remote)
```
here.

Reply:
Not using internals anymore. And I don't want to create method that is called only once.

Reply:
I don't understand either why are we exposing some internal utility method here. Repo _is_ API even if we decided to keep it hidden for now. It was made to be an API and we try to keep is clean. If there is a need for a simple mechanism to reuse certain helper, let's introduce _internal_ package or something.